artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -248,8 +255,13 @@ private ConnectionInfo getConnectionInfo() {
 
    //tells the connection that
    //some bytes just sent
-   public void bufferSent() {
-      lastSent = System.currentTimeMillis();
+   private void bufferSent() {
+      //much cheaper than a volatile set if contended, but less precise (ie allows stale loads)
+      LAST_SENT_UPDATER.lazySet(this, System.currentTimeMillis());
+   }
+
+   private static void traceBufferReceived(Object connectionID, Command command) {
```

Comment:
dedicated log message method, please add to the logger.

Reply:
I need to add a comment here: it is a trick to allow the caller to be inlined reducing bytecode size moving away uncommon paths...hence it doesn't have functional purposes.

Reply:
Still same can be done by using the logger.

Reply:
can't say if it has the same reason and could be addressed in that way: I do not need specific/common handling of those trace messages, but instead a way to packed them in separate methods leaving the callers smaller and able to be eligible for inlining.
Making them more generic just for perf reasons could be even worst wdyt?

Reply:
Its not generic, its still specific, its just following the conventions that specific logging lives in the Loggers, such as ActiveMQServerLogger (in this case maybe a new logger OpenWireLogger is needed/wanted)